### ISSUE TITLE
@prisma/adapter-d1 is failing with an import error when called inside vitest tests

### ISSUE THREAD

**Issue by mgcrea** [prisma/prisma#23911]
### Bug description

I'm trying to setup a small Cloudflare Worker app, that leverages `@prisma/adapter-d1` and served by [hono](https://hono.dev), I'm encountering issues when trying to run unit tests. Unfortunately the prisma WASM import fails with the following error:

```
Error: No such module "Users/olivier/Projects/cloudflare/cloudflare-hono-worker/node_modules/.prisma/client/#wasm-engine-loader".
    at Object.getQueryEngineWasmModule (Users/olivier/Projects/cloudflare/cloudflare-hono-worker/node_modules/.prisma/client/wasm.js?mf_vitest_no_cjs_esm_shim:182:5)
    at Users/olivier/Projects/cloudflare/cloudflare-hono-worker/node_modules/@prisma/client/runtime/wasm.js?mf_vitest_no_cjs_esm_shim:11:731
    at Object.loadLibrary (Users/olivier/Projects/cloudflare/cloudflare-hono-worker/node_modules/@prisma/client/runtime/wasm.js?mf_vitest_no_cjs_esm_shim:11:975)
    at gt.loadEngine (Users/olivier/Projects/cloudflare/cloudflare-hono-worker/node_modules/@prisma/client/runtime/wasm.js?mf_vitest_no_cjs_esm_shim:11:3194)
    at gt.instantiateLibrary (Users/olivier/Projects/cloudflare/cloudflare-hono-worker/node_modules/@prisma/client/runtime/wasm.js?mf_vitest_no_cjs_esm_shim:11:2778) {
  clientVersion: '5.12.1'
}
```
 

### How to reproduce

Create a basic d1 prisma app following the [official tutorial](https://www.prisma.io/docs/orm/prisma-client/deployment/edge/deploy-to-cloudflare#cloudflare-d1)

Add a vitest unit test:

```ts
import { createExecutionContext, env, waitOnExecutionContext } from "cloudflare:test";
import { testClient } from "hono/testing";
import { app } from "src/app";
import { describe, expect, test } from "vitest";

describe("App", () => {
  const testApp = testClient(app, env);

  test("GET /users", async () => {
    const ctx = createExecutionContext();
    const res = await testApp.users.$get({ ctx });
    await waitOnExecutionContext(ctx);
    expect(res.status).toBe(200);
    expect(await res.text()).toBe("Hello Hono!");
  });
});
```

Can be fixed with this change in `wasm.js`:

```js
config.engineWasm = {
  getRuntime: () => require('./query_engine_bg.js'),
  getQueryEngineWasmModule: async () => {
    return (await import('./query_engine_bg.wasm')).default // <-- fix
  }
}
```

but probably breaking other things!

### Expected behavior

Prisma should work inside vitest.

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: macOS
- Database: SQLite (D1 adapter)
- Node.js version: v20.12.0

### Prisma Version

```
▶ nr prisma -v                                                                                                                                    master|◒ 12 
10.5.0
```

**Comment by janpio** [comment]
Possibly related, also vitest, problem: https://github.com/prisma/prisma/issues/23855

**Comment by jkomyno** [comment]
Hi @mgcrea, thanks for opening this. While we reason about this issue internally, could you please share whether using the `vite-plugin-wasm` plugin in your `vitest` configuration would fix this problem without the need to modify `wasm.js`?

**Comment by Jolg42** [comment]
I could reproduce using the repro mentioned in:

              I also hit this bug (also mentioned https://github.com/cloudflare/workers-sdk/issues/5127#issuecomment-2063188923).

Here's a full reproduction repo https://github.com/vladinator1000/prisma-vitest-miniflare/tree/27410822df6161c2d97006352fe74dbd0692e90e

_Originally posted by @vladinator1000 in https://github.com/cloudflare/workers-sdk/issues/5685#issuecomment-2081836080_

```
npm i
npm run gen
npm run db-migrate 
npm run itest 


> prisma-vitest-miniflare@1.0.0 itest
> prisma generate && NODE_OPTIONS='--experimental-vm-modules' vitest --config ./vitest.config.integration.ts

Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma

✔ Generated Prisma Client (v5.12.1) to ./node_modules/@prisma/client in 114ms

....

 DEV  v1.3.0 /workspaces/prisma-vitest-miniflare

Using vars defined in .dev.vars
[vpw:inf] Starting isolated runtimes for vitest.config.integration.ts...
workerd/server/server.c++:2802: error: Fallback service failed to fetch module; payload = ; spec = /?specifier=%2Fworkspaces%2Fprisma-vitest-miniflare%2Fnode_modules%2F.prisma%2Fclient%2Fwasm-worker-loader.js&referrer=%2Fworkspaces%2Fprisma-vitest-miniflare%2Fnode_modules%2F.prisma%2Fclient%2Fwasm.js%3Fmf_vitest_no_cjs_esm_shim
 ❯ src/user.itest.ts (1)
   ❯ user model (1)
     × create user
       ⠼ [ beforeEach ]

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Failed Tests 1 ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

 FAIL  src/user.itest.ts > user model > create user
Error: No such module "workspaces/prisma-vitest-miniflare/node_modules/.prisma/client/#wasm-engine-loader".
 ❯ Object.getQueryEngineWasmModule workspaces/prisma-vitest-miniflare/node_modules/.prisma/client/wasm.js?mf_vitest_no_cjs_esm_shim:160:5
 ❯ workspaces/prisma-vitest-miniflare/node_modules/@prisma/client/runtime/wasm.js?mf_vitest_no_cjs_esm_shim:11:731
 ❯ Object.loadLibrary workspaces/prisma-vitest-miniflare/node_modules/@prisma/client/runtime/wasm.js?mf_vitest_no_cjs_esm_shim:11:975
 ❯ gt.loadEngine workspaces/prisma-vitest-miniflare/node_modules/@prisma/client/runtime/wasm.js?mf_vitest_no_cjs_esm_shim:11:3194
 ❯ gt.instantiateLibrary workspaces/prisma-vitest-miniflare/node_modules/@prisma/client/runtime/wasm.js?mf_vitest_no_cjs_esm_shim:11:2778

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
Serialized Error: { clientVersion: '5.12.1' }
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/1]⎯

 Test Files  1 failed (1)
      Tests  1 failed (1)
   Start at  14:08:05
   Duration  1.46s (transform 43ms, setup 0ms, collect 79ms, tests 25ms, environment 0ms, prepare 284ms)
```

**Comment by dwissaaj** [comment]
hey any update for this?

**Comment by jkomyno** [comment]
Hi @dwissaaj, this issue was fixed in https://github.com/prisma/prisma/pull/24554, which was released in Prisma 5.16.0.
